mlir/utils/jenkins/MIOpenDriver.py

```diff
@@ -97,35 +102,38 @@ def find_miopen_build_dir() -> str:
         if len(miopen_driver_path) == 0:
             # MIOpen driver not available
             return None
-        assert len(miopen_driver_path) == 1, "Multiple paths found to contain */bin/miopen-opt"
+        assert len(miopen_driver_path) == 1, "Multiple paths found to contain */bin/MIOpenDriver"
         miopen_driver_path = miopen_driver_path[0]
 
     miopen_build_dir = Path(miopen_driver_path).parent.parent
     return str(miopen_build_dir)
 
 def create_paths(mlir_build_dir_path, miopen_build_dir_path) -> Paths:
     """Creates the composite Paths structure using build dir paths"""
-
-    mlir_bin_dir = Path(mlir_build_dir_path) / 'bin'
-    mlir_lib_dir = Path(mlir_build_dir_path) / 'lib'
-    llvm_lib_dir = Path(mlir_build_dir_path) / 'external/llvm-project/llvm/lib'
-    configuration_dir = Path(mlir_build_dir_path) / '../mlir/utils/jenkins/miopen-tests'
-
-    miopen_driver_path = None
+    mlir_bin_dir = ''
+    mlir_lib_dir = ''
+    llvm_lib_dir = ''
+    if mlir_build_dir_path: 
+        mlir_bin_dir = str((Path(mlir_build_dir_path) / 'bin').resolve())
+        mlir_lib_dir = str((Path(mlir_build_dir_path) / 'lib').resolve())
+        llvm_lib_dir = str((Path(mlir_build_dir_path) / 'external/llvm-project/llvm/lib').resolve())
+
+    miopen_driver_path = ''
     if miopen_build_dir_path:
         miopen_driver_bin_dir = Path(miopen_build_dir_path) / 'bin'
         miopen_driver_path = str((miopen_driver_bin_dir / 'MIOpenDriver').resolve())
 
-    return Paths(
-        miopen_gen_path = str((mlir_bin_dir / 'miopen-gen').resolve()),
-        mlir_miopen_driver_path = str((mlir_bin_dir / 'mlir-miopen-driver').resolve()),
+    root_dir = str(subprocess.check_output(['git', 'rev-parse', '--show-toplevel']).decode().strip())
+    configuration_dir = root_dir + '/mlir/utils/jenkins/miopen-tests/resnet50-miopen-configs'
+
+    return Paths(miopen_gen_path = mlir_bin_dir + '/miopen-gen',
+        mlir_miopen_driver_path = mlir_bin_dir + '/mlir-miopen-driver',
         miopen_driver_path = miopen_driver_path,
-        rocm_runner_path = str((mlir_bin_dir / 'mlir-rocm-runner').resolve()),
-        libmlir_rocm_runtime_path =  str((llvm_lib_dir / 'libmlir_rocm_runtime.so').resolve()),
-        libconv_validation_wrappers_path = str((mlir_lib_dir / 'libconv-validation-wrappers.so').resolve()),
-        libmlir_runtime_utils_path = str((llvm_lib_dir / 'libmlir_runner_utils.so').resolve()),
-        configuration_file_path = str((configuration_dir / 'resnet50-miopen-configs').resolve()),
-    )
+        rocm_runner_path = mlir_bin_dir + '/mlir-rocm-runner',
```

Comment:
It seems to me like the nice thing to have here would be the function
```
def append_maybe_path(p: Optional[Path], subpart: str) -> Optional[str]:
  if path:
    return str((Path(p) / subpart).resolve())
  else:
    return None
```
and so we don't have paths like `"/libmlir_rocm_runtime.so"` running around the `Paths` object

Reply:
Agreed with the sentiment here and I'd also support having None rather than invalid paths running around.
I have some thoughts of some other structure that would suit the nature where mlir builds could also be optional as evident by this PR.